src/storage-controller/src/collection_mgmt.rs

```diff
@@ -1659,29 +1663,35 @@ where
-
-                // Duplicate rows ARE possible if many status changes happen in VERY quick succession,
-                // so we handle duplicated rows separately.
-                let entries = last_n_entries_per_key.entry(key).or_default();
-                for _ in 0..diff {
-                    // We CAN have multiple statuses (most likely Starting and Running) at the exact same
-                    // millisecond, depending on how the `health_operator` is scheduled.
-                    //
-                    // Note that these will be arbitrarily ordered, so a Starting event might
-                    // survive and a Running one won't. The next restart will remove the other,
-                    // so we don't bother being careful about it.
-                    //
-                    // TODO(guswynn): unpack these into health-status objects and use
-                    // their `Ord` impl.
-                    entries.push(Reverse((timestamp, row.clone())));
-
-                    // Retain some number of entries, using pop to mark the oldest entries for
-                    // deletion.
-                    while entries.len() > n {
-                        if let Some(Reverse((_, r))) = entries.pop() {
-                            deletions.push(r);
+            while let Some(chunk) = rows.next().await {
+                for ((key, _v), _t, diff) in chunk {
+                    let data = key.expect("successful decode");
+                    let Ok(row) = &data.0 else { continue };
+                    let (key, timestamp) = handle_row(row, diff);
+
+                    // Duplicate rows ARE possible if many status changes happen in VERY quick succession,
+                    // so we handle duplicated rows separately.
+                    let entries = last_n_entries_per_key.entry(key).or_default();
+                    for _ in 0..diff {
+                        // We CAN have multiple statuses (most likely Starting and Running) at the exact same
+                        // millisecond, depending on how the `health_operator` is scheduled.
+                        //
+                        // Note that these will be arbitrarily ordered, so a Starting event might
+                        // survive and a Running one won't. The next restart will remove the other,
+                        // so we don't bother being careful about it.
+                        //
+                        // TODO(guswynn): unpack these into health-status objects and use
+                        // their `Ord` impl.
+                        entries.push(Reverse((timestamp, row.clone())));
+
+                        // Retain some number of entries, using pop to mark the oldest entries for
+                        // deletion.
+                        while entries.len() > n {
+                            if let Some(Reverse((_, r))) = entries.pop() {
+                                let _ = deletions
```

Comment:
Is it OK to ignore the result of `add`? I'd expect a `expect` or so, to avoid silently eating the error. Or, maybe leave a comment to explain why its fine?